sdk/storage/Azure.Storage.DataMovement/src/TransferCheckpointer.cs

```diff
@@ -7,52 +7,108 @@
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
-using Azure.Core;
+using Azure.Storage.DataMovement.JobPlanModels;
 
 namespace Azure.Storage.DataMovement
 {
     /// <summary>
     /// Base Checkpointer class to create the checkpointing logic
     /// to resume from.
     /// </summary>
-    internal abstract class TransferCheckpointer
+    public abstract class TransferCheckpointer
     {
-        internal List<string> _ids;
-
         /// <summary>
         /// The protected constructor for the abstract TransferCheckpointer class (to allow for mocking).
         /// </summary>
-        protected TransferCheckpointer() { }
+        protected TransferCheckpointer()
```

Comment:
make a public localcheckpointer, leaves room for the other implementation of a checkpointer.

Reply:
Just to update on this comment, I think in our meeting we were moving towards making the checkpointer class private (or internal) and instead just taking in some form of options or other public type.

We _could_ think about leaving this abstract class public in case someone wanted to implement their own, but I also think that's probably not needed as I don't think anyone could implement one of these without extensive knowledge of the internals of DMLib.

Reply:
I'm making it internal for now. I think if there's a huge ask for customers to implement their own checkpointer then we can reconsider the decision.

It's easier to go from internal to public than it is to do public to internal. In the case if we also decide to make changes.